### PR TITLE
fix: tidy api of ec2 2004 creep and shrinkage

### PR DESCRIPTION
**Pull request by mortenengen**
An attempt to tidy the API of calculation of creep and shrinkage according to EC2 2004. This also depends on #193. The functions in this module do not represent only one equation each, but are in a few cases combinations of more than one equation. Since all the functions depend on time, it would also be great if an array of time values could be given as input.

Here is the plan:
- [x] Accept `ArrayLike` as input.
- [x] Merge dev when #193 is merged.
- [x] Factor out partial functions for shrinkage, such that all equations have separate functions, and drying shrinkage and autogenuous shrinkage can be separated.
- [x] Factor out partial functions for creep.
- [x] Homogenize docstrings.
- [x] Review public API including API docs.
- [x] Add tests to reach 100% coverage.
- [x] Rewrite maturity function `t_T` to better reflect the scope of Eq. (B.10).

### CODE REVIEW

**Comment by talledodiego** [comment]
Ops... You already wrote basically what I have mentioned in #192 and #193.

I agree with the idea.

**Review by talledodiego — CHANGES_REQUESTED** [review]
I am submitting this review as it is. Before going on I will wait to fixes on `np.isscalar` checks

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 1]
```diff
@@ -0,0 +1,507 @@
...
+
+    Args:
+        t (npt.ArrayLike): The age of the concrete in days.
+        t_s (float): The age of the concrete in days at the start of drying.
+        h_0 (float): The effective cross section thichkness in mm.
+
+    Returns:
+        npt.ArrayLike: The coefficient taking into account the time of drying.
+    """
+    t_drying = np.atleast_1d(t - t_s)
```
[the remark this answers is not in the document]
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py]: This is a very good point! I attempted to fix it in 5939a2d.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 2]
```diff
@@ -0,0 +1,507 @@
...
+        eps_cd_0 (float): The nominal value of drying shrinkage defined in Eq.
+            (B.11).
+
+    Returns:
+        npt.ArrayLike: The drying shrinkage.
+
+    Note:
+        In EC2 (2004), the shrinkage strain is calculated as a positive number.
+    """
+    return beta_ds * k_h * eps_cd_0
```
[the remark this answers is not in the document]
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py:62]: This is a very good point! I attempted to fix it in 5939a2d.

**Comment by mortenengen** [comment]
> I am submitting this review as it is. Before going on I will wait to fixes on `np.isscalar` checks

Thanks for the comment. This is a good one! I have added a fix. Please let me know if you agree with this.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+        fcm (float): The mean concrete strength in MPa.
+
+    Returns:
+        float: The effect of concrete strength.
+    """
+    return 16.8 / fcm**0.5
+
+
+def beta_t0(t0: float) -> float:
+    """Calculate the effect of age at loading on the standardized creep
```
Comment (talledodiego) [inline structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py:336]:
```suggestion
    """Calculate the effect of age at loading on the notional creep
```

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+    number.
```
Comment (talledodiego) [inline structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py:337]:
```suggestion
    coefficient.
```

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+"""
+
+from __future__ import annotations  # To have clean hints of ArrayLike in docs
+
+import typing as t
+
+import numpy as np
+import numpy.typing as npt
+
+ALPHA_CEMENT_DICT = {'R': 1.0, 'N': 0.0, 'S': -1.0}
```
Comment (talledodiego) [inline structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py:12]: Should we add a comment here indicating the reference to the code? Like defined in (B.9)?

**Review by talledodiego — CHANGES_REQUESTED** [review]
Excellent job! I find the code very solid! Just some minor comments.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py (hunk 5 above)
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py]: Good point. Fixed in a2dd665.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 6]
```diff
@@ -0,0 +1,507 @@
...
+ALPHA_DS_DICT = {
```
[the remark this answers is not in the document]
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py:13]: Good point. Fixed in a2dd665.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+
+    Returns:
+        float: The coefficient depending on the effective section thickness.
+    """
+    if h_0 >= 500:
+        k_h = 0.70
+    elif h_0 <= 100:
+        k_h = 1.0
+    else:
+        k_h = np.interp(h_0, [100, 200, 300, 500], [1.0, 0.85, 0.75, 0.7])
```
[the remark this answers is not in the document]
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py]: You are right, I simplified the function in b0d2591.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 8]
```diff
@@ -0,0 +1,507 @@
...
+
+
+def beta_ds(t: npt.ArrayLike, t_s: float, h_0: float):
+    """Calculate the coefficient taking into account the time of drying.
+
+    EN 1992-1-1:2004, Eq. (3.10).
+
+    Args:
+        t (npt.ArrayLike): The age of the concrete in days.
+        t_s (float): The age of the concrete in days at the start of drying.
```
[the remark this answers is not in the document]
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py]: I kind of agree, but one could also argue that technically, curing lasts as long as hydration goes (?). I slightly rewrote it in 96a0eb1.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 9]
```diff
@@ -0,0 +1,507 @@
...
+    Returns:
+        npt.ArrayLike: The coefficient taking into account the time of drying.
+    """
+    t_drying = np.atleast_1d(t - t_s)
+    t_drying[t_drying < 0.0] = 0.0
+    return t_drying / (t_drying + 0.04 * h_0 ** (3 / 2))
+
+
+def k_h(h_0: float) -> float:
+    """Calculate the coefficient depending on the effective section thickness.
```
[the remark this answers is not in the document]
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py]: Implemented in 96a0eb1.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 10]
```diff
@@ -0,0 +1,507 @@
...
+
+    EN 1992-1-1:2004, Tab. 3.3.
+
+    Args:
+        h_0 (float): The effective section thickness in mm.
```
[the remark this answers is not in the document]
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py]: Implemented in 96a0eb1.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py [hunk 11]
```diff
@@ -0,0 +1,507 @@
...
+
+    Returns:
+        float: The coefficient depending on the effective section thickness.
```
[the remark this answers is not in the document]
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py]: Implemented in 96a0eb1.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py (hunk 3 above)
Comment (mortenengen) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py]: Implemented in 96a0eb1.

**Comment by mortenengen** [comment]
Thanks for the comments. I have added changes accordingly. Please let me know what you think.

File: structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (talledodiego) [reply structuralcodes/codes/ec2_2004/_concrete_creep_and_shrinkage.py]: I see your point and I agree.

**Review by talledodiego — APPROVED** [review]

**Comment by talledodiego** [comment]
Outstanding! I think you did an excellent job here

**Pull request merged**